include/ginkgo/core/reorder/reordering_base.hpp

```diff
@@ -60,6 +60,12 @@ namespace reorder {
  * specific reordering to decide what to do with the data that is passed to it.
  */
 class ReorderingBase : public EnableAbstractPolymorphicObject<ReorderingBase> {
+public:
+    virtual std::shared_ptr<const gko::LinOp> get_permutation() const = 0;
+
+    virtual std::shared_ptr<const gko::LinOp> get_inverse_permutation()
```

Comment:
nit:
```suggestion
    virtual std::shared_ptr<const LinOp> get_permutation() const = 0;

    virtual std::shared_ptr<const LinOp> get_inverse_permutation()
```
Also note that this breaks the interface of RCM (as you had `std::shared_ptr<matrix::Permutation>` as the return type).